### ISSUE TITLE
PivotTable does not respect ColumnLabels order

### ISSUE THREAD

**Issue by UCFKnights83** [ClosedXML/ClosedXML#958]
## Read and complete the full issue template

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature

If you are logging a possible bug or feature request, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**

0.93.0.0

**What is the current behavior?**

PivotTable will not render ColumnLabels in the order in which they are added. It will keep the ColumnLabels in the order of the source Range.

**What is the expected behavior or new feature?**

PivotTable should maintain the order in which ColumnLabels are added.

**Did this work in previous versions of our tool?  Which versions?**

No

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

See code. Year should appear before Month based on the order being added to ColumnLabels but will only render in the order of the data Range, which has Year after Month.

**Code to reproduce problem:**
```c#
using ClosedXML.Excel;
using System;
using System.Collections.Generic;
using System.Data;
using System.IO;
using System.Linq;

namespace ClosedXMLPivotTable
{
    class Program
    {
        static void Main(string[] args)
        {
            DataTable dtTestData = new DataTable();
            dtTestData.Columns.Add("Search");
            dtTestData.Columns.Add("Search2");
            dtTestData.Columns.Add("Month");
            dtTestData.Columns.Add("Year");
            dtTestData.Columns.Add("Count");

            dtTestData.Rows.Add(new object[] { "Test", "Test", "Jan", 2018, 2 });
            dtTestData.Rows.Add(new object[] { "Test", "Test", "Feb", 2018, 5 });
            dtTestData.Rows.Add(new object[] { "Test", "Test", "Mar", 2018, 1 });
            dtTestData.Rows.Add(new object[] { "Test", "Test", "Apr", 2018, 12 });
            dtTestData.Rows.Add(new object[] { "Test", "Test", "May", 2018, 7 });
            dtTestData.Rows.Add(new object[] { "Test", "Test", "Jun", 2018, 4 });
            dtTestData.Rows.Add(new object[] { "Test", "Test", "Jul", 2018, 2 });
            dtTestData.Rows.Add(new object[] { "Test", "Test", "Aug", 2018, 3 });
            dtTestData.Rows.Add(new object[] { "Test2", "Test2", "Jan", 2018, 5 });
            dtTestData.Rows.Add(new object[] { "Test2", "Test2", "Feb", 2018, 14 });
            dtTestData.Rows.Add(new object[] { "Test2", "Test2", "Mar", 2018, 11 });
            dtTestData.Rows.Add(new object[] { "Test2", "Test2", "Apr", 2018, 2 });
            dtTestData.Rows.Add(new object[] { "Test2", "Test2", "May", 2018, 6 });
            dtTestData.Rows.Add(new object[] { "Test2", "Test2", "Jun", 2018, 6 });
            dtTestData.Rows.Add(new object[] { "Test2", "Test2", "Jul", 2018, 9 });
            dtTestData.Rows.Add(new object[] { "Test2", "Test2", "Aug", 2018, 3 });

            XLWorkbook wbWorkbook = new XLWorkbook();
            IXLWorksheet wsWorksheet = wbWorkbook.Worksheets.Add("TestData");
            
            List<Object[]> lstRowsToAdd = new List<Object[]>();
            lstRowsToAdd.Add(dtTestData.Columns.Cast<DataColumn>().Select(DC => DC.ColumnName).ToArray());

            dtTestData.Rows.Cast<DataRow>().ToList().ForEach(DRI =>
            {
                lstRowsToAdd.Add(DRI.ItemArray.Select(Item => Item != null ? Item : new Object()).ToArray());
            });

            wsWorksheet.Cell(wsWorksheet.RowsUsed().Count() + 1, 1).InsertData(lstRowsToAdd);

            IXLRange rRange = wsWorksheet.RangeUsed();
            IXLTable tTable = rRange.CreateTable();

            IXLWorksheet wsPivotTable = wbWorkbook.Worksheets.Add(String.Format("_{0}.PivotTable",
                wsWorksheet.RangeUsed().Worksheet.Name), wsWorksheet.RangeUsed().Worksheet.Position);
            IXLPivotTable ptPivotTable = wsPivotTable.PivotTables.Add(wsPivotTable.Name,
                wsPivotTable.Cell(1, 1), wsWorksheet.RangeUsed());

            IXLPivotField pfPivotField;
            IXLPivotValue pvPivotValue;

            pfPivotField = ptPivotTable.RowLabels.Add("Search");
            pfPivotField = ptPivotTable.RowLabels.Add("Search2");
            pfPivotField = ptPivotTable.ColumnLabels.Add("Year");
            pfPivotField = ptPivotTable.ColumnLabels.Add("Month");
            pvPivotValue = ptPivotTable.Values.Add("Count");

            wbWorkbook.SaveAs(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Desktop), "ClosedXMLPivotTable.xlsx"));
        }
    }
}
```
- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
[ClosedXMLPivotTable.xlsx](https://github.com/ClosedXML/ClosedXML/files/2220041/ClosedXMLPivotTable.xlsx)

**Comment by UCFKnights83** [comment]
Am I missing something to have this issue looked at? I see other issues opened after this one that are getting attention. I want to be sure I supplied everything that was needed.

**Comment by vbjay** [comment]
Pull requests are more than welcome.

**Comment by vbjay** [comment]
Everybody, Somebody, Anybody, and Nobody were members of a group.

There was an important job to do and Everybody was asked to do it.

Everybody was sure that Somebody would do it.

Anybody would have done it, but Nobody did it.

Somebody got angry because it was Everybody’s job.

Everybody thought Anybody would do it, but Nobody realized that Anybody
wouldn’t do it.

It ended up that Everybody, blamed Somebody, when Nobody did, what Anybody
could have done.

**Comment by igitur** [comment]
@UCFKnights83 Your original post seems fine. I'm just a bit tied up at work, and frankly, a bit tired of working on pivot tables for now. So other issues are more interesting to me. But as @vbjay says, if you want it fixed soon, submit a PR. This is a community effort.

**Comment by UCFKnights83** [comment]
@igitur Sorry, wasn't meant as a rush. I just wanted to be sure I had everything needed in the issue as I noticed others were being worked on. I fully understand how you feel about getting tired of a certain portions of a project. Some portions are either very daunting or are just a plain old pain in the butt. 

As a workaround, I went ahead and just reordered the columns of the DataTable to match the order of the the soon to be PivotTable.ColumnLabels order. Not ideal, but it definitely works now. I will submit a PR once I get some free time from my current project. Thanks!
